File: QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author sofiafernandezmoreno
 */
public class Calle extends Casilla{
    //Constructor para TipoCasillas que son calle
    private TituloPropiedad titulo;
    
    Calle (int numCasilla, TituloPropiedad tp){
        super(numCasilla, tp.getPrecioCompra());
        titulo = tp;
    }
    
    public TituloPropiedad asignarPropietario(Jugador jugador){
        titulo.setPropietario(jugador);
        return titulo;
    }
    
    @Override
    protected TipoCasilla getTipo(){
        return TipoCasilla.CALLE;
    }

    @Override
    protected TituloPropiedad getTitulo(){
        return titulo;
    }

   
    public boolean tengoPropietario(){
        return titulo.tengoPropietario();
    }
    
    public int pagarAlquiler(){
        int costeAlquiler = titulo.pagarAlquiler();
        return costeAlquiler;
    }
    
    private void setTitulo(TituloPropiedad titulo) {
        this.titulo = titulo;
    }

    @Override
    protected boolean soyEdificable(){
        return true;
    }
    
    @Override
    public String toString(){
        return super.toString() + "\n - Tipo: CALLE\n - Titulo:\n" + titulo;
    }

    
}

File: QytetetRuby/lib/calle.rb
#encoding: utf-8
module ModeloQytetet
  require_relative "casilla"

  class Calle < Casilla
    attr_reader :titulo
    def initialize(numeroCasilla, titulo)
      super(numeroCasilla, titulo.precioCompra,TipoCasilla::CALLE) 
      @titulo = titulo
    end
    
    def asignarPropietario(jugador)
      @titulo.propietario = jugador
      return @titulo
    end
    def pagarAlquiler
      costeAlquiler = @titulo.pagarAlquiler

      return costeAlquiler
    end

    def soyEdificable
      return true
    end

    def tengoPropietario
      @titulo.tengoPropietario
    end
    
    def to_s
      texto = super.to_s
      texto += "Titulo: "
      
      unless @titulo == nil
        texto << "#{@titulo}"
      end
        
      texto
    end
  end
end
